### PR TITLE
Teacher view /time auto start

### PR DESCRIPTION
**Pull request by solaolateju**
**Teacher View**
Screenshot 2020-05-19 at 20 32 32

**Start button visible, when auto start is off**
Screenshot 2020-05-19 at 20 45 32

### CODE REVIEW

File: src/components/modes/teacher/TeacherView.js [hunk 1]
```diff
@@ -10,10 +11,12 @@ import {
   patchAppInstanceResource,
   postAppInstanceResource,
   deleteAppInstanceResource,
+  getUsers,
   openSettings,
 } from '../../../actions';
-import { getUsers } from '../../../actions/users';
+// import { getUsers } from '../../../actions/users';
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:17]: You should delete this.

File: src/components/modes/teacher/Settings.js [hunk 2]
```diff
@@ -180,6 +205,17 @@ class Settings extends Component {
           )}
           label={t('Show Time Controls to Students')}
         />
+        <Tooltip title="Starting automatically will remove student controls">
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Settings.js:208]: Remember to change the tooltip title:

"If the timer does not start automatically and there are no controls, the student will still be able to see a start button."

Also, wrap in `t()` function.

File: src/components/modes/teacher/Responses.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                      onClick={this.handleToggleConfirmDialog(true)}
+                      disabled={_.isEmpty(userContent)}
+                    >
+                      <DeleteIcon />
+                    </IconButton>
+                    <ConfirmDialog
+                      open={confirmDialogOpen}
+                      title={t('Delete Time')}
+                      text={t(
+                        "By clicking 'Delete', you will be deleting students' time. This action cannot be undone.",
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Responses.js:130]: `the student's`

File: src/components/modes/teacher/Responses.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    });
+  };
+
+  handleConfirmDelete = id => {
+    const { dispatchDeleteAppInstanceResource } = this.props;
+    dispatchDeleteAppInstanceResource(id);
+    this.handleToggleConfirmDialog(false)();
+  };
+
+  milliToHMS = duration => {
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/Responses.js:68]: I would rename this `formatTime` and describe in a comment:

`// takes millis and outputs hh:mm:ss`

File: src/components/modes/student/StudentView.js [hunk 5]
```diff
@@ -55,12 +58,20 @@ export const StudentView = props => {
   const timeResourceId = timeResource && (timeResource.id || timeResource._id);
   const timeResourceData = timeResourceId ? timeResource.data : null;
 
+  const startAuto = startImmediately || started;
+
+  // if (!timeControlsVisible)
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentView.js:63]: Remove this.

File: src/components/modes/student/StudentMode.js [hunk 6]
```diff
@@ -4,30 +4,28 @@ import { connect } from 'react-redux';
...
+
+  componentDidMount() {
+    const { userId, dispatchGetAppInstanceResources } = this.props;
 
     // get the resources for this user
-    props.dispatchGetAppInstanceResources({ userId });
+    dispatchGetAppInstanceResources({ userId });
   }
 
   componentDidUpdate({ appInstanceId: prevAppInstanceId }) {
```
Comment (juancarlosfarah) [inline src/components/modes/student/StudentMode.js:31]: Can you change this to:

```
  componentDidUpdate({ appInstanceId: prevAppInstanceId, userId: prevUserId }) {
    const {
      appInstanceId,
      dispatchGetAppInstanceResources,
      userId,
    } = this.props;
    // handle receiving the app instance id
    if (appInstanceId !== prevAppInstanceId || userId !== prevUserId) {
      dispatchGetAppInstanceResources({ userId });
    }
  }
```

To see if it fixes the bug we introduced, where the saved time was not loaded on refresh. Thanks!

Comment (solaolateju) [reply src/components/modes/student/StudentMode.js:31]: No it did not fix the bug. The saved time did not load on refresh

**Pull request merged**